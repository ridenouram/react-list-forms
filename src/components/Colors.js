import React from 'react';
import Color from './Color';
import PropTypes from 'prop-types';

export default function Colors({ colors }) {

  const colorList = colors.map(({ name, hex, rgb }) => {
    return (
      <li key={hex}>
        <Color name={name} hex={hex} rgb={rgb} />
      </li>
    );
  });

  return (
    <ul>
      {colorList}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};
