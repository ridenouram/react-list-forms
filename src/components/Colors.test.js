import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('Colors component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Colors colors={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
