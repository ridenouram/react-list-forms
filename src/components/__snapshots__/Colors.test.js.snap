// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component matches snapshot 1`] = `<ul />`;
